File: 3/pract-03-03.cpp
#include <iostream>
#include <cmath>

// Функция для вычисления члена ряда
double term(int k) {
    return 4.0 / pow(2, k - 1) + pow(-1, k - 1) / (2 * pow(3, k - 1));
}

// Главная функция
int main() {
    const double epsilon = 1e-6;
    double sum = 0.0;
    int k = 1;

    while (true) {
        double t = term(k);
        sum += t;
        if (std::fabs(t) < epsilon) break;
        ++k;
    }

    std::cout << "Sum of the series: " << sum << std::endl;
    return 0;
}

File: 3/pract-03-03.py
import math

# Функция для вычисления члена ряда
def term(k):
    return 4.0 / (2**(k - 1)) + (-1)**(k - 1) / (2 * 3**(k - 1))

# Главная функция
if __name__ == "__main__":
    epsilon = 1e-6
    sum = 0.0
    k = 1

    while True:
        t = term(k)
        sum += t
        if abs(t) < epsilon:
            break
        k += 1

    print("Sum of the series: {:.6f}".format(sum))
